File: data/readable_format/fortran/file289.f90
module TransferBug
    implicit none

    type :: ByteType
        character(len=1) :: singleByte
    end type ByteType

    type (ByteType), allocatable :: BytesPrototype(:)

contains

    function StringToBytes(v) result(bytes)
        character(len=*), intent(in) :: v
        type(ByteType), allocatable :: bytes(:)
        integer :: i

        allocate(bytes(len(v)))
        do i = 1, len(v)
            bytes(i)%singleByte = v(i:i)
        end do
    end function StringToBytes

    subroutine BytesToString(bytes, string)
        type(ByteType), intent(in) :: bytes(:)
        character(len=*), intent(out) :: string
        integer :: numChars, i

        numChars = size(bytes)
        string = ''
        do i = 1, numChars
            string(i:i) = bytes(i)%singleByte
        end do
    end subroutine BytesToString

    subroutine TestConversion()
        character(len=100) :: testStr = "Hello, world!"
        character(len=100) :: resultStr
        call BytesToString(StringToBytes(testStr), resultStr)
        if (trim(resultStr) /= trim(testStr)) then
            print *, "Test Failed: ", trim(testStr), " != ", trim(resultStr)
        else
            print *, "Test Passed: ", trim(testStr), " = ", trim(resultStr)
        end if
    end subroutine TestConversion

end module TransferBug

program main
    use TransferBug
    implicit none

    call TestConversion()

end program main

File: data/readable_format/cpp/file289.cpp
#include <iostream>
#include <vector>
#include <string>

class ByteType {
    char singleByte;
public:
    ByteType() : singleByte(0) {}
    ByteType(char c) : singleByte(c) {}
    char getByte() const { return singleByte; }
};

std::vector<ByteType> StringToBytes(const std::string &v) {
    std::vector<ByteType> bytes;
    for (char c : v) {
        bytes.push_back(ByteType(c));
    }
    return bytes;
}

std::string BytesToString(const std::vector<ByteType> &bytes) {
    std::string str;
    for (const ByteType &b : bytes) {
        str += b.getByte();
    }
    return str;
}

// Simple testing mechanism
void TestConversion(const std::string& testStr) {
    auto bytes = StringToBytes(testStr);
    std::string convertedStr = BytesToString(bytes);
    if (convertedStr == testStr) {
        std::cout << "Test Passed: " << testStr << " == " << convertedStr << std::endl;
    } else {
        std::cerr << "Test Failed: " << testStr << " != " << convertedStr << std::endl;
    }
}

int main() {
    // Test cases
    TestConversion("Hello, world!");
    TestConversion("Test string with multiple characters, numbers 1234567890, and symbols !@#$%^&*()");

    return 0;
}